### PR TITLE
Fix for josdk 1179

### PR DESCRIPTION
**Pull request by scrocquesel**
In the reconciler, if the secret is not yet ready, an event should be fired when it will be the case.
The reconciler should not throw on purpose to retry but let new events the opportunity to reconcile

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -11,6 +11,11 @@
   <version>4.0.0.Beta2-SNAPSHOT</version>
   <packaging>pom</packaging>
   <name>Quarkus - Operator SDK - Parent</name>
+  <properties>
+    <quarkus.version>2.8.2.Final</quarkus.version>
+    <java-operator-sdk.version>3.0.0.RC3</java-operator-sdk.version>
+    <kubernetes-client.version>5.12.2</kubernetes-client.version>
+  </properties>
```
Comment (metacosm) [inline pom.xml:18]: Not sure about this change… This make it easier to deal from our side but wouldn't that make it harder to consume the BOM from a user's perspective?

**Comment by metacosm** [comment]
Also, can you please update that PR so that it targets the `updates` branch, instead? Thanks in advance!

File: pom.xml (hunk 1 above)
Comment (scrocquesel) [reply pom.xml:18]: I confirm it still works

**Review by metacosm — APPROVED** [review]
LGTM

**Pull request merged**